### PR TITLE
Refs Task#387252 #387249 Delete payload: Coding + Unit test

### PR DESCRIPTION
**Pull request by tamtt-1031**
## Related Tickets

- [#387252](https://dev.sun-asterisk.com/issues/387252)
- [#387249](https://dev.sun-asterisk.com/issues/387249)

## What's this PR do ?

- [x] Delete payload: Coding
- [x] Delete payload: Unit test

## Library
*(List package, library third party add new include version)*

- N/A

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*
Screenshot from 2019-12-02 11-29-16

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/PayloadController.php [hunk 1]
```diff
@@ -100,8 +100,15 @@ public function update(Request $request, Payload $payload)
...
+    public function destroy(Webhook $webhook, Payload $payload)
     {
-        //
+        try {
+            $this->payloadRepository->delete($payload->id);
+
+            return redirect()->route('webhooks.edit', $webhook)
+                             ->with('messageSuccess', 'This payload successfully deleted');
+        } catch (Exception $exception) {
+            return redirect()->back()->with('messageFail','Delete failed. Something went wrong');
```
Comment (framgiaciv3) [inline app/Http/Controllers/PayloadController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Functions.FunctionCallArgumentSpacing.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadController.php</br>**Line**: 111</br>No space found after comma in function call

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/PayloadController.php [hunk 2]
```diff
@@ -100,8 +100,15 @@ public function update(Request $request, Payload $payload)
...
-    public function destroy(Payload $payload)
+    public function destroy(Webhook $webhook, Payload $payload)
     {
-        //
+        try {
+            $this->payloadRepository->delete($payload->id);
+
+            return redirect()->route('webhooks.edit', $webhook)
+                             ->with('messageSuccess', 'This payload successfully deleted');
+        } catch (Exception $exception) {
```
Comment (quangvv-1620) [inline app/Http/Controllers/PayloadController.php]: bắt luôn exception thì hơi căng á a

**Review by ghost — APPROVED** [review]

File: app/Http/Controllers/PayloadController.php (hunk 2 above)
Comment (quangvv-1620) [reply app/Http/Controllers/PayloadController.php]: bữa e làm bên ruby thấy họ bảo ko nên bắt cả exception, ko biết bên này có bị chi ko
https://stackoverflow.com/a/10048406

**Review by ghost — APPROVED** [review]

**Review by nguyenthinhhn — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**